src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -9,12 +10,680 @@ public class PropertiesShouldNotBeWriteOnlyTests : DiagnosticAnalyzerTestBase
+{
+    public class ClassWithReadableProperty
+    {
+        protected string field;
+        public string AccessibleProperty1
+        {
+            get { return field; }
+            private set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class ClassWithReadableProperty
+    {
+        protected string field;
+        protected string AccessibleProperty2
+        {
+        get { return field; }
+        private set { field = value; }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_internal_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        internal string AccessibleProperty3
+        {
+            get { return field; }
+            private set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_internal_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        protected internal string AccessibleProperty4
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty6
+        {
+            get { return field; }
+            protected set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    protected string field;
+    public class DerivedClassWithReadableProperty : ClassWithReadableProperty
+    {
```

Comment:
This test is broken because ClassWithReadable property is not defined in this snippet. Please include that as well.